src/gamelib/SDL/bitmap.cpp

```diff
@@ -60,10 +60,18 @@ void BmpFile::SaveToFile(const ColorBitmap & data, std::string_view file_name)
 template <typename BitmapType, typename RawDataType, typename RawDataDecodeFunc>
 BitmapType BmpFile::LoadFromFile(std::string_view file_name, RawDataDecodeFunc DecodeFunc)
 {
-    const auto native_surface = std::unique_ptr<SDL_Surface, void (*)(SDL_Surface *)>(
-        SDL_LoadBMP(file_name.data()), [](SDL_Surface * surface) { SDL_FreeSurface(surface); });
+    /* zero size SDL_Deleted will give you sizeof(std::unique_ptr<T, Deleter>) == sizeof(void *) */
+    
+    struct SDL_Deleter {
```

Comment:
capture-less lambda zabira vic? Zero-size je zajimave, ale zas to cele zabira uz dost radek na takovou blbost :)

Reply:
Ale asi to muzu udelat pak jako globalni typ a prestane to byt takovy psaci overhead

Reply:
ano, zabírá více, protože dáváš typ deleteru jako funkční pointer, takže unique_ptr jí uloží dovnitř unique_ptr jako ten pointer, když máš deleter typ, tak ho použije, ale vyoptimalizuje to že je nulový

Reply:
pokud budeš mít více SDL věcí na uvolnění, tak doporučuji to mít globálně s pár různýma overloadama a pak mít `template <typename T> using sdl_unique_ptr = std::unique_ptr<T, SDL_deleter>;`